### PR TITLE
Add implementation for ws_value_int type

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #69

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Note: Our specification says we should implement this one with 64 bit. I implemented it using 32 bit. What should be used?

File: src/values/int.c [hunk 1]
```diff
@@ -25,5 +25,104 @@
...
+
+/**
+ * Deinitialize callback for ws_value_int type
+ */
+static void
+int_deinit(
+    struct ws_value* self
+) {
+    if (self) {
+        if (self->type == WS_VALUE_TYPE_INT) {
```
Comment (neithernut) [inline src/values/int.c]: You don't need this check.

File: src/values/int.c [hunk 2]
```diff
@@ -25,5 +25,104 @@
...
+    struct ws_value_int* other
+) {
+    if (self && other) {
+        if (self->value.type == WS_VALUE_TYPE_INT &&
+                other->value.type == WS_VALUE_TYPE_INT) {
+
+            signed int res;
+
+            pthread_rwlock_rdlock(&self->i.rwl);
+            pthread_rwlock_rdlock(&other->i.rwl);
```
Comment (neithernut) [inline src/values/int.c]: You created a potential deadlock.
Use the `try` variant and potentially a condition or something.

Comment (matthiasbeyer) [inline src/values/int.c]: I guess a `try` lock wouldn't be beneficial here, but I don't know how to properly use conditions (by now).

Would you care to add a patch on this branch? If not, you are welcome to teach me how to implement it tomorrow evening!

Comment (neithernut) [inline src/values/int.c]: It'll be tomorrow evening then.

Comment (neithernut) [inline src/values/int.c]: As discussed somewhere else, remove the locks completely.

**Comment by matthiasbeyer** [comment]
I'm ready here, ACK for squashing?

File: src/values/int.c [hunk 3]
```diff
@@ -25,5 +25,87 @@
...
+
 #include "values/int.h"
+#include "values/value.h"
+#include "util/attributes.h"
+
+/**
+ * Deinitialize callback for ws_value_int type
+ */
+static void
+int_deinit(
```
Comment (neithernut) [inline src/values/int.c]: Set the deinit callback to ``NULL and be done with it. No need fo a NOP function.

**Comment by neithernut** [comment]
GO for squashing.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**